src/consensus/aft/raft.h

```diff
@@ -456,13 +454,8 @@ namespace aft
       assert(new_learner_nodes.empty());
 
 #ifdef CCF_RAFT_TRACING
-      nlohmann::json j = {};
-      j["function"] = "add_configuration";
-      j["state"] = *state;
-      j["configurations"] = configurations;
-      j["new_configuration"] = Configuration{idx, conf, idx};
-      j["committable_indices"] = committable_indices;
-      RAFT_TRACE_JSON_OUT(j);
+      trace::Line line("add_configuration", state, configurations);
+      line.j["new_configuration"] = Configuration{idx, conf, idx};
```

Comment:
This will need a local block as soon as we do scenarios with retirement, because of line 471.